src/main/java/com/example/TicketBookingApp/service/ReservationService.java

```diff
@@ -0,0 +1,139 @@
+                .screening(screening)
+                .personName(reservationRequest.getPersonName())
+                .personSurname(reservationRequest.getPersonSurname())
+                .expirationTime(screening.getStartTime().plus(Duration.ofMinutes(30)).atDate(screening.getDate()))
+                .build();
+
+        reservation.setTickets(reservationRequest
+                .getTicketRequests()
+                .stream()
+                .map(ticketRequest -> createTicketForReservation(ticketRequest, reservation))
+                .collect(Collectors.toList()));
+
+
+        reservation.setTotalAmountToPay((float) reservation
+                .getTickets()
+                .stream()
+                .mapToDouble(Ticket::getTicketPrice)
+                .sum()
+        );
+
+        LocalDateTime localDateTime = LocalDateTime.now();
+        LocalDateTime screeningDateTime = LocalDateTime.of(screening.getDate(), screening.getStartTime());
+
+        if (Duration.between(localDateTime, screeningDateTime).toMinutes() > MAX_TIME_BEFORE_START_SCREENING) {
+            reservationRepository.save(reservation);
+            return ReservationResponse
+                    .builder()
+                    .movieTitle(reservation.getScreening().getMovie().getTitle())
+                    .date(reservation.getDate())
+                    .time(reservation.getTime())
+                    .personName(reservation.getPersonName())
+                    .personSurname(reservation.getPersonSurname())
+                    .expirationTime(reservation.getExpirationTime())
+                    .numberOfTickets(reservation.getTickets().size())
+                    .totalAmountToPay(reservation.getTotalAmountToPay())
+                    .roomNumber(reservation.getScreening().getScreeningRoom().getRoomNumber())
+                    .build();
+        } else
+            throw new BookingTimeException();
+    }
+
+    private Boolean checkAvailabilityOfAdjacentSeat(Seat seat) {
+        Seat secondSeatFromTheLeft = seatRepository
+                .findByScreeningRoom_RoomNumberAndRowNumberAndSeatNumber(seat.getScreeningRoom().getRoomNumber(), seat.getRowNumber(), seat.getSeatNumber() - DISTANCE_OF_TWO_PLACES);
+        Seat leftAdjacentSeat = seatRepository
+                .findByScreeningRoom_RoomNumberAndRowNumberAndSeatNumber(seat.getScreeningRoom().getRoomNumber(), seat.getRowNumber(), seat.getSeatNumber() - DISTANCE_OF_ONE_PLACE);
+        Seat secondSeatFromTheRight = seatRepository
+                .findByScreeningRoom_RoomNumberAndRowNumberAndSeatNumber(seat.getScreeningRoom().getRoomNumber(), seat.getRowNumber(), seat.getSeatNumber() + DISTANCE_OF_TWO_PLACES);
+        Seat rightAdjacentSeat = seatRepository
+                .findByScreeningRoom_RoomNumberAndRowNumberAndSeatNumber(seat.getScreeningRoom().getRoomNumber(), seat.getRowNumber(), seat.getSeatNumber() + DISTANCE_OF_ONE_PLACE);
+
+        if (!ObjectUtils.isEmpty(secondSeatFromTheLeft)) {
```

Comment:
wow... `isEmpty` jest bardzo bogate, ale czy nie lepiej by było z repozytorium zwrócić Optionala i użyć jego metod?

Reply:
Racja. Lepiej byłoby to sprawdzić metodą _isPresent()_